flink-kubernetes/src/main/java/org/apache/flink/kubernetes/entrypoint/KubernetesApplicationClusterEntrypoint.java

```diff
@@ -146,14 +148,15 @@ private static ArtifactFetchManager.Result fetchArtifacts(Configuration configur
             String targetDir = generateJarDir(configuration);
             ArtifactFetchManager fetchMgr = new ArtifactFetchManager(configuration, targetDir);
 
-            List<String> uris = configuration.get(PipelineOptions.JARS);
-            checkArgument(uris.size() == 1, "Should only have one jar");
+            List<String> uris =
+                    configuration.getOptional(PipelineOptions.JARS).orElse(Collections.emptyList());
+            checkArgument(uris.size() <= 1, "Should only have at most one jar");
```

Comment:
How about handling the jar file requirements for SQL jobs in the same way as we do for Python jobs? 
For example, by introducing a method like PackagedProgramUtils.isSqlScript(String entryPointClassName)?
And doing the same for yarn as well?

Reply:
I think SQL Program is much similar to the Jar Program rather than Python Program: 
* Python uses zip or py files to run the application, but SQL Program uses jar to run the applicaiton. 
* Python Program has its own dependencies management through python.files, but SQL Program prefers to use deployment to upload files and use the PackagedProgram to build the required environment.

Considering SQL Program is much like Jar Prograom, do we need to add a if-else branch for SQL Program, which may introduce some duplicate codes.

On the other hand, I think it's possible that some users may put some jars in the image. The change can also reduce the cost for users to play with application mode(they don't need to upload files every time).

Reply:
Ok, I understand your point. However, currently, SQL jobs should rely on artifacts for jar dependencies, and there is no need for a pipeline.jar. 
But on second thought, the issue with my initial proposal is that it's not appropriate for the runtime side to determine the concept of SQL. 
So considering everything, I think maintaining the existing logic is acceptable.